pkg/api/analysisrun/key_test.go

```diff
@@ -1,24 +1,27 @@
-package pkgidentifier
+package analysisrun_test
```

Comment:
We don't have tests in a separate package elsewhere in the project. Is this something specific to the `pkg/` folder?

Reply:
As per https://pkg.go.dev/testing a `_test` suffixed package is a way to ensure that the tests are only testing the exported interface (i.e. black box testing).

This is my general preference as it encourages testable interfaces, makes refactoring easier as tests don't depend on implementation details, ensures the interface works as expected, and provides clear examples of how to use the interface.

Not using it is more an artifact of how the project has grown, with unit-tests not being there from the start.

Reply:
Fair enough